### ISSUE TITLE
change build status to an actual enum in the postgres build summary table

### ISSUE THREAD

**Issue by shankj3** [shankj3/ocelot#55]
right now we deducing the build status off of the various times that have been added 
```golang 
failed_validation := statuses.BuildSum.BuildTime.Seconds == 0 && statuses.BuildSum.BuildTime.Nanos == 0 && statuses.BuildSum.QueueTime.Seconds == 0
queued := statuses.BuildSum.BuildTime.Seconds == 0 && statuses.BuildSum.BuildTime.Nanos == 0 && statuses.BuildSum.QueueTime.Seconds > 0
buildStarted := statuses.BuildSum.BuildTime.Seconds > 0 && statuses.IsInConsul
finished := !statuses.IsInConsul && statuses.BuildSum.BuildTime.Seconds > 0
```

this is just not working out, its inflexible  

just add a column in `build_summary`: `build_status` that reflects the enum that we already try to deduce on the client side here in `client/commandhelper/table.go`:

```golang 

const (
	RUNNING BuildStatus = iota
	QUEUED
	DONE
	FAILED_PRESTART
)

//queued := statuses.BuildSum.BuildTime.Seconds == 0 && statuses.BuildSum.BuildTime.Nanos == 0
//buildStarted := statuses.BuildSum.BuildTime.Seconds > 0 && statuses.IsInConsul
//finished := !statuses.IsInConsul && buildStarted
func GetStatus(queued, buildStarted, finished, failed_validation bool) BuildStatus {
	if queued {
		return QUEUED
	}
	if buildStarted {
		return RUNNING
	}
	if finished {
		return DONE
	}
	if failed_validation {
		return FAILED_PRESTART
	}
	panic("none of these!")
}

```

---
so we can start with those 4 states and get rid of this crap. the downside is that you're gonna have a column that will, except for the active builds, all be the same value... 
which i really don't like either
idk

**Comment by shankj3** [comment]
nvm we should just change the "failed" column to a status column then the enum would be 
```golang
const (
	RUNNING BuildStatus = iota
	QUEUED
	FAILED
        PASSED
	FAILED_PRESTART
)
```

**Comment by shankj3** [comment]
# also do this for build_stage_details status

**Comment by shankj3** [comment]
closed  by #71
